Improvements to readability

AlexxNica opened the pull request:
Also updated the license year.

AlexxNica commented:
Just calling @ChadKillingsworth and @MatrixFrog to review this. Thanks! 🤗

ChadKillingsworth commented:
I've looked at it - just hadn't had a chance to respond.

A lot of the changes are simply stylistic and not substantial. For instance, we don't need to remove the line breaks - long lines are hard to read in many editors.

Can you modify the PR to only include things that are wrong? Capitalization of proper names would be such a case. If there are sections that are grammatically incorrect, that's fine to change too.

AlexxNica commented:
@ChadKillingsworth **Thanks for taking the time to comment on this.**
I haven't found any removed line breaks as you said; instead, the only line break changes I see are the ones I added to make lists of options, for example, where they were written using too many words and all upon the same line, making it much harder to understand. There's where I changed some lines to contain lists of options, instead of just regular text.
Making it a lot easier for users with [or without] disabilities to read and understand.

Take screen readers and official guidelines as an example:

**Here is the Office of Administration Federal Standards for Interface Accessibility** (_only for lists, as that's what we're talking about here_):
https://oa.mo.gov/sites/default/files/CC-Accessibility-508FedStds-Lists041205.pdf

You'll find the following statements right in the first lines:  

> **Description:**
> Lists refer to groups of related items." [...] "Lists can be used to help provide logical flow of content.
> 
> **Rationale:**
> [...] How screen readers render lists needs to be considered. [...]
>
> **Benefits:**
> Using standard practices for lists provides the following benefits:
> - Better access when using screen readers.
> - Reduces technology barriers for individuals with disabilities.
> - Results in greater customer satisfaction as well as increased use.

**Also following the [Web Content Accessibility Guidelines (WCAG) 2.0 [WCAG20]](https://www.w3.org/TR/WCAG20/) on the technique "H48: Using ol, ul and dl for lists or groups of links":**
https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/H48

**Or even statements of the [Community College of Denver (CCD)](https://www.ccd.edu/), enforcing the [Colorado Community College System (CCCS) President's Procedure 3-125g (Web Accessibility Procedure)](https://www.cccs.edu/sp-3-125g-web-accessibility-procedure/):**
> **LISTS**
> It is important to format a list as a list.
>
> Formatting elements like lists, headings and links, are read aloud
> to screen reader users, so the content is understood in context.

Reference for the above:
https://www.ccd.edu/administration/non-academic-departments/teaching-learning-center/accessibility-wcag#tabs-5

or for direct access (**_beware that this will download the PDF automatically_**):
https://www.ccd.edu/download/file/fid/6760

If that's not what you were talking about, could you please point which lines you think should be changed?

I understand that most changes might seem as "simply stylistic and not substantial," but we do need to think about accessibility and readability, as it's one of the most important things to get more people into the project. Both for using or contributing to it.

Again, I appreciate the time you're taking to consider this.

ChadKillingsworth commented on README.md:
```diff
@@ -1,32 +1,28 @@
-sure to read and understand the
-[compilation levels](https://developers.google.com/closure/compiler/docs/compilation_levels) as
-the compiler works very differently depending on the compilation level selected.
+If you are new to [Closure-Compiler](https://developers.google.com/closure/compiler/), make sure to read and understand the [compilation levels](https://developers.google.com/closure/compiler/docs/compilation_levels), as the compiler works very differently depending on selected level.
```
This is a case where line breaks were removed.

ChadKillingsworth commented on README.md:
Line break

ChadKillingsworth commented on README.md:
This phrasing is now not correct. `build tool plugins` is correct.

ChadKillingsworth commented on README.md:
```diff
@@ -37,20 +33,18 @@ npm install --save google-closure-compiler
-The compiler has a large number of flags. The best documentation for the flags can be found by
-running the `--help` command of the compiler.jar found inside the
-`node_modules/google-closure-compiler` folder:
+The compiler has a large number of flags, all of which you can see by running the compiler.jar file (inside `node_modules/google-closure-compiler` folder) with the `--help` command, like so:
```
Line breaks

ChadKillingsworth commented on README.md:
```diff
@@ -61,8 +55,8 @@ Values are either strings or booleans. Options which have multiple values can be
-For advanced usages, the options may be specified as an array of strings. These values _include_
-the "--" characters and are directly passed to the compiler in the order specified:
+An array of strings can be used to specify your options if advanced usage is needed.
+In this case, options keys *must* be preceded by `--`, and the compiler processes them in the specified order (left to right, top to bottom).
```
This sentence is just slightly different phrasing of what was already there.

ChadKillingsworth commented on README.md:
```diff
@@ -74,11 +68,9 @@ the "--" characters and are directly passed to the compiler in the order specifi
 
-When an array of flags is passed, the input files should not be specified via the build tools, but
-rather as compilation flags directly.
+- Input files should not be specified via build tools when passing an array of option flags. Instead, you should use compilation flags as you would, directly.
```
This was reworded so now the primary indicator (Array of flags) is much later in the sentence. This is an unnecessary change.

ChadKillingsworth commented on README.md:
```diff
@@ -98,8 +90,7 @@ require('google-closure-compiler').grunt(grunt);
 
-Task targets, files and options may be specified according to the grunt
-[Configuring tasks](http://gruntjs.com/configuring-tasks) guide.
+* Options, files, and task targets may be specified according to Grunt's [Configuring Tasks Guide](http://gruntjs.com/configuring-tasks).
```
This rewording is just a different style. Not needed.

ChadKillingsworth commented on README.md:
```diff
@@ -307,8 +295,11 @@ var compilerProcess = closureCompiler.run(function(exitCode, stdOut, stdErr) {
+
 ## License
-Copyright 2015 The Closure Compiler Authors
+Copyright © 2017 The Closure Compiler Authors
```
The license is legal text. This can't be changed in any way without approval.

ChadKillingsworth commented on README.md:
````diff
@@ -307,8 +295,11 @@ var compilerProcess = closureCompiler.run(function(exitCode, stdOut, stdErr) {
 ```
 
+## Version History
+Closure Compiler release notes can be found on the [main repository wiki](https://github.com/google/closure-compiler/wiki/Binary-Downloads).
````
It's fine to move this, but the line break needs to stay.

ChadKillingsworth commented:
While I admire your desire to make the web a more accessible place, this content was accessible as is. The items you called out are to make sure when using lists, use them semantically. Meaning, don't type numbers and bullets. This readme already did that.  It is not a directive to create a list where one didn't exist beforehand.

That being said, converting things to a list wasn't really the problem. The PR changed almost every line of the README. In many cases it was only whitespace changes. In others, it was simply a different phrasing for the same sentence. These are the types of changes that should be avoided.

AlexxNica commented:
I've had a problem with my Mac, so I'll be posting to this PR within the next couple of days.

@ChadKillingsworth, again, I really appreciate the time you're taking to review this. :+1:

ChadKillingsworth commented:
Closing this for inactivity.
